Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -61,10 +67,11 @@ async function run() {
         console.log(tl.loc('GotconnectiondetailsforazureRMWebApp0', webAppName));
 
         // For container based linux deployment
-        if(webAppKind && webAppKind.indexOf("linux") !== -1 && dockerNamespace)
-        {
+        if(!isWindowsWebApp && (!imageSource || imageSource.indexOf("Builtin") < 0) && dockerNamespace) {
             tl.debug("Performing container based deployment.");
-
+            if (webAppUri) {
```

Comment:
Can we move this out of the condition? 
We need to set the App URL for all the App Types

Reply:
We are setting it for all App Types. In case of webApps other than Linux, we tend to update the destination Url if the option virtual Application is checked. So I have updated it here explicitly for Linux Webapps